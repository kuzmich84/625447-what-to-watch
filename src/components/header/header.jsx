import React from "react";
import {getAuthorisationStatus, getAvavtar} from "../../store/selectors";
import {connect} from "react-redux";
import {AuthorisationStatus} from "../../const";
import PropTypes from "prop-types";
import {Link} from "react-router-dom";

const Header = ({authorisationStatus, avatar}) => {

  return (
    <header className="page-header movie-card__head">
      <div className="logo">
        <a className="logo__link">
          <span className="logo__letter logo__letter--1">W</span>
          <span className="logo__letter logo__letter--2">T</span>
          <span className="logo__letter logo__letter--3">W</span>
        </a>
      </div>

      <div className="user-block">
        {authorisationStatus === AuthorisationStatus.AUTH
          ? <div className="user-block__avatar">
            <img src={avatar} alt="User avatar" width="63" height="63"/>
          </div>
          : <Link to="/login" className="page-title">Sign in</Link>
        }

      </div>
    </header>
  );
};

const mapStateToProps = (state) => ({
  authorisationStatus: getAuthorisationStatus(state),
  avatar: getAvavtar(state),
});

export default connect(mapStateToProps)(Header);

Header.propTypes = {
  authorisationStatus: PropTypes.string.isRequired,
  avatar: PropTypes.string,
};
