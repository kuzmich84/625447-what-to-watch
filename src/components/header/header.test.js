import React from "react";
import renderer from "react-test-renderer";
import {Header} from "./header";
import {BrowserRouter as Router} from 'react-router-dom';

test(`Should Header render correctly`, () => {
  const tree = renderer
    .create(<Router>
      <Header
        authorisationStatus={`AUTH`}
        avatar={`img.jpg`}
        page={`main`}
      /> </Router>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
