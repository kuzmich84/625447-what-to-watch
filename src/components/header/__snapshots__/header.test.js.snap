// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Header render correctly 1`] = `
Array [
  <header
    className="page-header movie-card__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href={null}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="user-block"
    >
      <div
        className="user-block__avatar"
      >
        <a
          href="/mylist"
          onClick={[Function]}
        >
          <img
            alt="User avatar"
            height="63"
            src="img.jpg"
            width="63"
          />
        </a>
      </div>
    </div>
  </header>,
  " ",
]
`;
